.github/workflows/ScreenShotTest.yml

```diff
@@ -43,15 +43,52 @@ jobs:
         if: ${{ hashFiles('preview-screenshots/out/failures/*.png') != '' }}
         uses: actions/upload-artifact@v3
         with:
-          name: scrennshot-test-results
+          name: screenshot-test-results
           path: preview-screenshots/out/failures
 
-      - name: Comment PR
+      - name: Checkout the docs branch
+        id: checkout_docs
         if: ${{ hashFiles('preview-screenshots/out/failures/*.png') != '' }}
+        uses: actions/checkout@v3
+        with:
+          path: temp_docs
+          ref: docs
+
+      - name: Commit the screenshot to the branch
+        id: docs_images
+        if: steps.checkout_docs.outcome == 'success'
         continue-on-error: true # https://github.com/DroidKaigi/conference-app-2022/issues/497
+        run: |
+          cp -a preview-screenshots/out/failures/. temp_docs/docs/screenshots/
```

Comment:
I have a question 🙋 If there are multiple PRs updating images, will the other PR's image change?

Reply:
Forgot to add that to the description, yes it might be one issue. 

Combining with your comment about using Zipline, I guess we need to consider the "companion branch" approach instead. The overall idea would be the same, but we need to do more work:

- Create new branch from the workflow.
- Delete the branch after the PR is merged.

Let me give it a look and update this PR in a couple days 🙇 .